sentry_streams/sentry_streams/pipeline/pipeline.py

```diff
@@ -431,3 +452,47 @@ class FlatMap(TransformStep[Any]):
     """
 
     step_type: StepType = StepType.FLAT_MAP
+
+
+######################
+# Complex Primitives #
+######################
+
+
+@dataclass
+class Parser(Step, Generic[TransformFuncReturnType]):
```

Comment:
Is this the version that incorporates the changes we discussed or are you still working on this ?
I do not see the method to produce the basic primitives.